FoodOrderingSystem.py

```diff
@@ -0,0 +1,700 @@
+            print("Error getting menu, error={}".format(str(ex)))
+        finally:
+            session.close()            
+
+    def create_order_id(self, cust_id, session): 
+        """ Generate order id """
+        order = CustOrderStatus(cust_id=cust_id)
+        session.add(order)
+        try:
+            session.commit()
+            return order
+        except:
+            session.rollback()
+            raise Exception("Order ID not created!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def add_food_to_order(self, order_id, food_id, food_qty, session): 
+        """ Add food items """
+        add = CustOrderSelection(order_id=order_id, food_id=food_id, food_qty=food_qty)
+        session.add(add)
+        try:
+            session.commit()            
+            return add
+        except:
+            session.rollback()
+            raise Exception("Food items not added!")
+        finally:
+            session.expunge_all()
+            session.close()
+        
+    def view_order(self, cust_id, session):
+        """ Customer can view their orders before checkout/confirming order """
+        try:
+            view = session.query(FoodCategory, FoodDetails, CustOrderSelection, CustomerDetails, CustOrderStatus).\
+                filter(CustomerDetails.cust_id == CustOrderStatus.cust_id).\
+                filter(CustOrderSelection.food_id == FoodDetails.food_id).\
+                filter(FoodCategory.category_id == FoodDetails.category_id).\
+                filter(CustOrderSelection.order_id == CustOrderStatus.order_id).\
+                filter(CustOrderStatus.cust_id == cust_id)
+                # label('total_price'), func.sum(CustOrderSelection.food_qty*FoodDetails.price)
+                # (func.sum(CustOrderSelection.food_qty*FoodDetails.price).label("total_price"))
+            return view
+        except Exception as ex:
+            print("Error getting order, error={}".format(str(ex)))
+        finally:
+            session.close()
+
+    def checkout(self, cust_id, order_status, order_address, checkout_time, estimated_time, session):
+        """ Customer can checkout/confirm order """
+        checkout_update = session.query(
+            CustOrderStatus
+            ).filter(CustOrderStatus.cust_id == cust_id
+            ).update(
+                {CustOrderStatus.order_status: order_status, 
+                CustOrderStatus.order_address: order_address, 
+                CustOrderStatus.checkout_time: checkout_time,
+                CustOrderStatus.estimated_time: estimated_time}, 
+                synchronize_session=False
+                )
+        try:
+            session.commit()
+            return checkout_update
+        except:
+            session.rollback()
+            raise Exception("Checkout not completed!")
+        finally:
+            session.expunge_all()
+            session.close()
+
+    def delete_order(self, order_id, session):
```

Comment:
delete_order() should not delete the order but update the order with the status as "DELETED", instead how about using a cancel_order() function instead and update the status to CANCELLED.

Reply:
Change made.

When customer wants to delete an order, it will update the CustOrderStatus table as cancelled.